editor/src/clj/editor/outline_view.clj

```diff
@@ -40,159 +49,234 @@
+                            :node-outline-key-path node-outline-key-path
+                            :parent-reference? parent-reference?
+                            :children (mapv :outline data)
+                            :child-error (boolean (coll/some :child-error data))
+                            :child-overridden (boolean (coll/some :child-overridden data))
+                            :hideable (contains? outline-name-paths (rest node-outline-key-path))
+                            :hidden-parent (scene-visibility/hidden-outline-key-path? hidden-node-outline-key-paths node-outline-key-path)
+                            :scene-visibility (if (contains? hidden-node-outline-key-paths node-outline-key-path)
+                                                :hidden
+                                                :visible))]
+              {:outline outline
+               :child-error (or (:child-error outline) (:outline-error? outline))
+               :child-overridden (or (:child-overridden outline) (:outline-overridden? outline))}))]
+    (let [outline (:outline (decorate outline [] [] (:outline-reference? outline)))
+          node-id->node-id-paths (util/group-into {} [] key val
+                                                  (coll/transfer [outline] :eduction
+                                                    (coll/tree-xf :children :children)
+                                                    (map (coll/pair-fn :node-id :node-id-path))))
+          default-selection (coll/transfer selection {}
+                              (keep #(when-let [node-id-paths (node-id->node-id-paths %)]
+                                       (coll/pair % #{(first node-id-paths)}))))
+          full-expanded (->> node-id->node-id-paths
+                             (e/mapcat val)
+                             (e/mapcat #(e/take-while coll/not-empty (iterate pop %)))
+                             (reduce #(assoc! %1 %2 false) (transient {}))
+                             persistent!)
+          default-expanded (coll/merge
+                             full-expanded
+                             (-> default-selection
+                                 auto-expand
+                                 (assoc (:node-id-path outline) true)))]
+      {:outline outline
+       :default-selected-node-id->node-id-paths default-selection
+       :default-node-id-path->expanded default-expanded})))
+
+(def ^:private ext-with-tree-view-props
+  (fx/make-ext-with-props
+    (assoc fx.tree-view/props
+      :selected-node-id-paths
+      (fx.prop/make
+        (fx.mutator/setter
+          (fn set-selected-node-id-paths [^TreeView tree-view new-selected-node-id-paths]
+            (let [selection-model (.getSelectionModel tree-view)
+                  old-selected-node-id-paths (coll/transfer (.getSelectedItems selection-model) #{}
+                                               (keep #(:node-id-path (.getValue ^TreeItem %))))]
```

Comment:
Not sure if it matters here, but the order of the selected items does have an effect on which item the Scene View gizmo is placed on, so using a set might not be entirely correct. 🤔